app/multitenant/dynamo_collector.go

```diff
@@ -141,6 +156,25 @@ func NewDynamoDBCollector(
 		}
 	}
 
+	var memcacheClient *memcache.Client
+	if memcachedHost != "" {
+		var memcacheServers memcache.ServerList
+		memcacheClient = memcache.NewFromSelector(&memcacheServers)
+		memcacheClient.Timeout = memcachedTimeout
+		err := updateMemcacheServers(memcachedHost, memcachedService, &memcacheServers)
+		if err != nil {
+			// REVIEWER: Undecided whether this should exit or not. Thoughts?
+			log.Errorf("Could not set memcache servers: %v", err)
+		}
+
+		ticker := time.NewTicker(memcacheUpdateInterval)
+		go func() {
+			for range ticker.C {
+				updateMemcacheServers(memcachedHost, memcachedService, &memcacheServers)
+			}
+		}()
```

Comment:
I'd like to encapsulate this behaviour in a separate file / type please, much like we do with the consul client.